app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt

```diff
@@ -309,20 +311,39 @@ private fun assertSearchEngineURL(searchEngineName: String) {
         .check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
 }
 
-private fun assertSearchEngineResults(rule: ComposeTestRule, searchSuggestion: String, searchEngineName: String) {
-    rule.waitUntil(waitingTime, waitForSearchSuggestions(rule, searchSuggestion, searchEngineName))
+private fun assertSearchEngineResults(rule: ComposeTestRule, searchSuggestion: String) {
+    rule.waitUntil(waitingTime, waitForSearchSuggestions(rule, searchSuggestion))
     rule.onNodeWithText(searchSuggestion).assertIsDisplayed()
 }
 
-private fun waitForSearchSuggestions(rule: ComposeTestRule, searchSuggestion: String, searchEngineName: String): () -> Boolean =
+private fun waitForSearchSuggestions(rule: ComposeTestRule, searchSuggestion: String): () -> Boolean =
     {
         rule.waitForIdle()
         mDevice.waitForObjects(mDevice.findObject(UiSelector().textContains(searchSuggestion)))
-        rule.onAllNodesWithTag("mozac.awesomebar.suggestion").assertAny(hasText(searchSuggestion) and hasText(searchEngineName))
+        rule.onAllNodesWithTag("mozac.awesomebar.suggestion").assertAny(hasText(searchSuggestion))
         mDevice.findObject(UiSelector().textContains(searchSuggestion)).waitForExists(waitingTime)
     }
 
-private fun assertSearchEngineSuggestionResults(rule: ComposeTestRule, searchResult: String) {
+private fun assertDefaultSearchEngineSuggestionResults(rule: ComposeTestRule, searchSuggestion: String, searchTerm: String) {
+    var currentTries = 0
+    while (currentTries++ < 3) {
+        try {
+            rule.waitUntil(waitingTime, waitForSearchSuggestions(rule, searchSuggestion))
+            rule.onNodeWithText(searchSuggestion).assertIsDisplayed()
+
+            break
+        } catch (e: AssertionError) {
+            homeScreen {
```

Comment:
Just to verify that this works, have you tried throwing `AssertionError` in the try block?

Reply:
Yes, it's throwing `AssertionError`

But it looks that I overlooked to check if the test fails with `AssertionError` if the suggestions isn't found the 3rd time. (right know the test will always pass because it will always be able to execute `typeSearch(searchTerm)` from the catch block)

I'll make some changes and re-run it.